hack/infra/ksnap-acr.bicep

```diff
@@ -15,10 +15,11 @@ resource acr 'Microsoft.ContainerRegistry/registries@2023-11-01-preview' = {
 
 var schedule = '0 1 * * Tue' // 1am UTC every Tuesday
 
+// --untagged appears to be broken: https://github.com/Azure/acr-cli/issues/131
 var purgeOldArtifacts = '''
 version: v1.1.0
 steps:
-- cmd: acr purge --filter 'karpenter/snapshot/.*:.*' --ago 30d --untagged
+- cmd: acr purge --filter 'karpenter/snapshot/.*:.*' --ago 30d
```

Comment:
this will remove any image that is older than 30 days for snapshot images? 

I also read about dry run we could use to test some of this if we wanted to add tests for this kind of thing. 

--dry-run - Specifies that no data is deleted, but the output is the same as if the command is run without this flag. This parameter is useful for testing a purge command to make sure it does not inadvertently delete data you intend to preserve.

Reply:
> this will remove any image that is older than 30 days for snapshot images?

Yes

> --dry-run

Yes, though I'd prefer to check the results of the actual execution instead; don't quite trust `--dry-run` at this point